### PR TITLE
Introduce IsRunning and IsPublishing properties.

### PR DESCRIPTION
**Pull request by mitchdenny**
As discussed in our conversation about AppInsights/resource provisioning.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2372)

### CODE REVIEW

File: src/Aspire.Hosting/DistributedApplicationExecutionContext.cs [hunk 1]
```diff
@@ -13,4 +13,14 @@ public class DistributedApplicationExecutionContext(DistributedApplicationOperat
...
+
+    /// <summary>
+    /// Returns true if the current operation is publishing.
+    /// </summary>
+    public bool IsPublishing => Operation == DistributedApplicationOperation.Publish;
+
+    /// <summary>
+    /// Returns true if the current operation is running.
+    /// </summary>
+    public bool IsRunning => Operation == DistributedApplicationOperation.Publish;
```
Comment (eerhardt) [inline src/Aspire.Hosting/DistributedApplicationExecutionContext.cs:25]: "IsRunning" is kind of a weird name. The app isn't actually running yet.

"IsRunContext" ?

[the remark this answers is not in the document]
Comment (mitchdenny) [reply src/Aspire.Hosting/DistributedApplicationExecutionContext.cs:25]: haha thanks.

Comment (mitchdenny) [reply src/Aspire.Hosting/DistributedApplicationExecutionContext.cs]: IsRunMode / IsPublishMode?

Comment (eerhardt) [reply src/Aspire.Hosting/DistributedApplicationExecutionContext.cs]: I like IsRunMode more than IsRunning.

Comment (davidfowl) [reply src/Aspire.Hosting/DistributedApplicationExecutionContext.cs]: Mode hmmmm, don't love it but....

Comment (mitchdenny) [reply src/Aspire.Hosting/DistributedApplicationExecutionContext.cs]: `IsRunning` that bad? I don't have a strong opinion here. Although I find it weird that we have `.Operation` but a `xMode` suffix on the boolean properties.

**Review by davidfowl — APPROVED** [review]

**Pull request merged**